android/app/src/main/java/org/electroncash/electroncash3/ColdLoad.kt

```diff
@@ -0,0 +1,86 @@
+package org.electroncash.electroncash3
+
+import android.content.ClipData
+import android.content.ClipboardManager
+import android.content.Intent
+import android.text.Editable
+import android.text.TextWatcher
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.ViewModel
+import com.google.zxing.integration.android.IntentIntegrator
+import kotlinx.android.synthetic.main.load.*
+import kotlinx.android.synthetic.main.main.*
+
+// This provides a dialog to allow users to input a string, which is then broadcast
+// on the bitcoin cash network. Strings are not validated, but broadcast_transaction2 should throw error which is toasted.
+// Valid transaction quickly show up in transactions.
+
+class ColdLoadDialog : AlertDialogFragment() {
+
+    class Model : ViewModel() {}
+
+    val model: Model by viewModels()
+
+    override fun onBuildDialog(builder: AlertDialog.Builder) {
+        builder.setTitle(R.string.load_transaction)
+                .setView(R.layout.load)
+                .setNegativeButton(android.R.string.cancel, null)
+                .setNeutralButton(R.string.qr_code, null)
+                .setPositiveButton(R.string.send, null)
+    }
+
+    override fun onShowDialog() {
+        super.onShowDialog()
+        val ourClipboard = getSystemService(ClipboardManager::class)
+
+        etTransaction.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
+            override fun afterTextChanged(s: Editable?) {
+                val currenttext = etTransaction.text
+                //checks if text is blank. further validations can be added here
+                dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = currenttext.isNotBlank()
+            }
+
+        })
+        dialog.getButton(AlertDialog.BUTTON_POSITIVE).setOnClickListener { onOK() }
+        dialog.getButton(AlertDialog.BUTTON_NEUTRAL).setOnClickListener { scanQR(this) }
+        dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = false
+        btnPaste.isEnabled = ourClipboard.hasPrimaryClip()
+        btnPaste.setOnClickListener {
+            val clipdata = ourClipboard.primaryClip
+            val cliptext = clipdata!!.getItemAt(0)
```

Comment:
If the user switches to another app which clears the clipboard, and then switches back to Electron Cash, then `primaryClip` will be null and this line will crash. Better to just leave the button enabled all the time, and if there's nothing on the clipboard then clicking it should do nothing.

Reply:
removed